### PR TITLE
Chore/workflow improvements

### PR DESCRIPTION
**Pull request by samwelkanda**
Related https://github.com/microsoftgraph/msgraph-sdk-python/issues/519
* Add default templates for bugs and features
* Add Python 3.12 and separate build and publish workflows
* Add workflows to create release and publish package
* Add templates for pull requests and releases

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - name: Release Notes
+        run: |
+          git log $(git describe HEAD~ --tags --abbrev=0)..HEAD --pretty='format:* %h %s%n' --no-merges >> ".github/release_template.md"
```
Comment (baywet) [inline .github/workflows/release.yml:24]: again, we have a changelog we maintain manually at this stage, and which contains a cleaned-up version of the changes (this is goin got get a lot of noise), why not use that instead?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    environment: pypi_prod
+    needs: [build]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
```
Comment (baywet) [inline .github/workflows/publish.yml:24]:
```suggestion
          python-version: 3.12
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  autorelease:
+    name: Create release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.11
```
[the remark this answers is not in the document]
Comment (samwelkanda) [reply .github/workflows/release.yml]: Updated to 3.12

File: .github/workflows/release.yml (hunk 1 above)
Comment (samwelkanda) [reply .github/workflows/release.yml]: Updated the workflow to use an action that extracts release notes from changelog when creating a release
https://github.com/marketplace/actions/extract-release-notes

Comment (baywet) [reply .github/workflows/release.yml]: oooh this is even better better than the script! thanks for sharing this! Resolving

**Review by baywet — APPROVED** [review]

**Pull request merged**